toucan_connectors/toucan_connector.py

```diff
@@ -426,13 +426,12 @@ def get_unique_identifier(self) -> str:
         """
         return self.json()
 
-    def _render_datasource(self, data_source: ToucanDataSource) -> dict:
-        data_source_rendered = nosql_apply_parameters_to_query(
-            data_source.dict(), data_source.parameters, handle_errors=True
-        )
-        del data_source_rendered['parameters']
-
-        return data_source_rendered
+    def _get_unique_datasource_identifier(self, data_source: ToucanDataSource) -> dict:
+        # By default we don't know which variable syntax is be supported by the inheriting connector,
+        # so calling `nosql_apply_parameters_to_query` is wrong and will produce the same cache key
+        # for different queries when using custom variable syntax !
+        # Overwrite this method to improve the cache key at places where supported syntaxes are clear.
```

Comment:
:fire: thanks for this docstring !

Reply:
Bets to use `"""` for doctrings :) (IDEs and doc generators can use doc strings defined with triple quotes, but not single-line comments)